internal/queue/consumer.go

```diff
@@ -59,17 +64,63 @@ func (q *RabbitQueue) StartConsumer(queueData ConsumerQueue, worker worker.Worke
 	}
 	go func() {
 		for message := range messages {
-			err := worker.Process(message.Body)
-			if err != nil {
-				err = message.Acknowledger.Nack(message.DeliveryTag, false, true)
+			switch message.RoutingKey {
+			case ActorRoutingKey:
+				aw := worker.ActorWorker{
+					Logger:     q.logger,
+					Repository: repository.NewRepository(conn),
+				}
+
+				err = aw.Process(message.Body)
+				if err != nil {
+					err = message.Acknowledger.Nack(message.DeliveryTag, false, true)
+					if err != nil {
+						panic(err) // I know this isn't good. Will need to fix it
+					}
+					continue
+				}
+				err = message.Acknowledger.Ack(message.DeliveryTag, false)
 				if err != nil {
 					panic(err) // I know this isn't good. Will need to fix it
 				}
-				continue
-			}
-			err = message.Acknowledger.Ack(message.DeliveryTag, false)
-			if err != nil {
-				panic(err) // I know this isn't good. Will need to fix it
+			case CommentRoutingKey:
+				aw := worker.CommentWorker{
+					Logger:     q.logger,
+					Repository: repository.NewRepository(conn),
+				}
+
+				err = aw.Process(message.Body)
+				if err != nil {
+					err = message.Acknowledger.Nack(message.DeliveryTag, false, true)
+					if err != nil {
+						panic(err) // I know this isn't good. Will need to fix it
+					}
+					continue
+				}
+				err = message.Acknowledger.Ack(message.DeliveryTag, false)
+				if err != nil {
+					panic(err) // I know this isn't good. Will need to fix it
+				}
+			case PostRoutingKey:
+				aw := worker.PostWorker{
```

Comment:
We shouldn't be re-instantiating things inside of a loop. Instead, we should instantiate them outside the loop and then reference them inside it. This will reduce risk of memory leak as well as improve the performance of not needing to constantly be reconnecting to the database